src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Interfaces/ICrazyPriceRepository.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using Exadel.CrazyPrice.Common.Models.Request;
+using Exadel.CrazyPrice.Common.Models.Response;
+
+namespace Exadel.CrazyPrice.Common.Interfaces
+{
+    public interface ICrazyPriceRepository
+    {        
+
+        /// <summary>
+        /// Gets all Discount Programs.
+        /// </summary>
+        /// <returns></returns>
+        Task<IEnumerable<DiscountResponse>> GetAllDiscountProgramsAsync();
```

Comment:
Probably List<> will be better? IEnumerable may cause delayed LINQ queries with unpredictive results

Reply:
[Fix it](https://github.com/yegorl/exadel_discounts_be/blob/fix/src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Interfaces/IDiscountRepository.cs)